File: JavaParser/src/Target.java
public class Target {
	private String strings;
	Target(String target_strings) {
		this.strings = target_strings;
	}

	public String getStrings() {
		return this.strings;
	}

	public boolean isEqual(Target comparison) {
		if (this.strings.equals(comparison.strings)) {
			return true;
		}
		else {
			return false;
		}
	}

	public boolean isBigger(Target comparison) {
		if (Integer.parseInt(this.strings) >= Integer.parseInt(comparison.strings)) {
			return true;
		}
		else {
			return false;
		}
	}

	public Target extract(int lastIndex, int index) {
		return new Target(this.strings.substring(lastIndex, index));
	}
}

File: PHPParser/Target.php
<?php

class Target {
	private $strings;

	public function __construct($target) {
		$this->strings = $target;
	}

	public function getStrings() {
		return $this->strings;
	}

	public function isEqual($comparison) {
		if ($this->strings === $comparison->strings) {
			return true;
		}
		else {
			return false;
		}
	}

	public function isBigger($comparison_obj) {
		if ($this->strings >= $comparison_obj->strings) {
			return true;
		}
		else {
			return false;
		}
	}

	public function extract($index, $chara_num = 1) {
		return new Target(substr($this->strings, $index, $chara_num));
	}
}